Update pseudo elements page (#27907)
mdn/content: files/en-us/web/css/pseudo-elements/index.md

title: Pseudo-elements
slug: Web/CSS/Pseudo-elements
page-type: landing-page
spec-urls:
  - https://drafts.csswg.org/css-pseudo/
  - https://drafts.csswg.org/css-position-4/
  - https://drafts.csswg.org/css-shadow-parts/
  - https://w3c.github.io/webvtt/
---

{{CSSRef}}

> A CSS **pseudo-element** is a keyword added to a selector that lets you style a specific part of the selected element(s). 

Author: ```suggestion
A CSS **pseudo-element** is a keyword added to a selector that lets you style a specific part of the selected element(s).
```

}
```

> ## Syntax

Reviewer: Thank you. I agree, we can remove the example and just keep the syntax. And maybe do the same on Pseudo-classes page.

}
```

Double colons (`::`) are used for pseudo-elements. This distinguishes pseudo-classes from pseudo-elements where single colon (`:`) is used.

> You can use only one pseudo-element in a selector. That single pseudo-element must appear after all the other components in any [complex](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) or [compound](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#compound_selector) selector in which it appears. For example, you can select a paragraph's first line with `p::first-line`, but cannot select first-line's children or a hovered first line. Both `p::first-line > *` and `p::first-line:hover` are invalid. While it is not possible to style a pseudo-element based on its _state_, they can be styled based on the state of the element in which they are found: `p:hover::first-line` is valid.

Reviewer: - Can we add to why those examples "are invalid" following from the previous explanation?
- I've split the paragraph here to separate the two thoughts.
- I hope the edit to the second para is technically accurate.

```suggestion
You can use only one pseudo-element in a selector. The pseudo-element must appear after all the other components in the [complex](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) or [compound](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#compound_selector) selector in which it appears. For example, you can select a paragraph's first line using `p::first-line` but not the first-line's children or a hovered first line. So both `p::first-line > *` and `p::first-line:hover` are invalid because...

While it is not possible to select an element based on its _state_ by using pseudo-elements, a pseudo-element can be used to select and style a part of an element that already has a state applied to it. For example, `p:hover::first-line` selects the first line (pseudo-element) of a paragraph when the paragraph itself is being hovered (pseudo-class).
```

> It is important to note, when a selector list contains an invalid selector, the entire style block is ignored.

Reviewer: should we put this in a note?

```suggestion
> **Note:** When a [selector list](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#selector_list) contains an invalid selector, the entire style block is ignored.
```

> ## Alphabetical index

Reviewer: how about renaming this to

```suggestion
## List of pseudo-elements
```

}
```

Double colons (`::`) are used for pseudo-elements. This distinguishes pseudo-elements from [pseudo-classes](/en-US/docs/Web/CSS/Pseudo-classes) that use single colon (`:`) in their notation.

> You can use only one pseudo-element in a selector. The pseudo-element must appear after all the other components in the [complex](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) or [compound](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#compound_selector) selector in which it appears. For example, you can select a paragraph's first line using `p::first-line` but not the first-line's children or a hovered first line. So both `p::first-line > *` and `p::first-line:hover` are invalid because...

Reviewer: ```suggestion
You can use only one pseudo-element in a selector. The pseudo-element must appear after all the other components in the [complex](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) or [compound](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#compound_selector) selector in which it appears. For example, you can select a paragraph's first line using `p::first-line` but not the first-line's children or a hovered first line. So both `p::first-line > *` and `p::first-line:hover` are invalid.
```

- {{CSSxRef("::target-text")}} {{Experimental_Inline}}

> > **Note:** Browsers support single colon syntax for the original four pseudo-elements, `::before`, `::after`, `::first-line`, and `::first-letter`, only.

Reviewer: ```suggestion
> **Note:** Browsers support the single colon syntax only for the original four pseudo-elements: `::before`, `::after`, `::first-line`, and `::first-letter`.
```

## Specifications

{{Specifications}}

## See also

> - [CSS pseudo-element](/en-US/docs/Web/CSS/CSS_pseudo) module

Reviewer: Curious - would the slug for the incoming module page not have 'element', that is, CSS_pseudo-elements? Also, the nit with the name "CSS pseudo-elements" module

Author: the spec is https://drafts.csswg.org/css-pseudo/, so i followed that.

Reviewer: Yeah, there are multiple pages with -elements slug and even though we internally might be able to differentiate between a landing page and guide, and it can be confusing for readers, especially because currently there is not much hierarchy under CSS folders. I guess something to keep in mind for future about these selector pages.

- [Pseudo-classes](/en-US/docs/Web/CSS/Pseudo-classes)
- [CSS selectors](/en-US/docs/Web/CSS/CSS_selectors) module

> - [Building block: pseudo-classes and pseudo-elements](/en-US/docs/Learn/CSS/Building_blocks/Selectors/Pseudo-classes_and_pseudo-elements)

Reviewer: ```suggestion
- [CSS building blocks: Pseudo-classes and pseudo-elements](/en-US/docs/Learn/CSS/Building_blocks/Selectors/Pseudo-classes_and_pseudo-elements)
```
